src/TerritorialCouncil/Filter/MembersListFilter.php

```diff
@@ -0,0 +1,150 @@
+<?php
+
+namespace App\TerritorialCouncil\Filter;
+
+use App\Entity\ReferentTag;
+use Symfony\Component\Validator\Constraints as Assert;
+
+class MembersListFilter
+{
+    /**
+     * @var ReferentTag[]
+     *
+     * @Assert\NotNull
+     */
+    private $referentTags = [];
+
+    /**
+     * @var bool|null
+     */
+    private $emailSubscription;
+
+    /**
+     * @var string|null
+     *
+     * @Assert\Length(max=255)
+     */
+    private $firstName;
+
+    /**
+     * @var string|null
+     *
+     * @Assert\Length(max=255)
+     */
+    private $lastName;
+
+    /**
+     * @var string|null
+     */
+    private $gender;
+
+    /**
+     * @var string
+     *
+     * @Assert\NotBlank
+     * @Assert\Choice(choices={"lastName"})
+     */
+    private $sort = 'adherent.lastName';
+
+    /**
+     * @var string
+     *
+     * @Assert\NotBlank
+     * @Assert\Choice(choices={"a", "d"})
+     */
+    private $order = 'a';
+
+    public function __construct(array $referentTags, string $subscriptionType)
+    {
+        $this->referentTags = $referentTags;
+        $this->subscriptionType = $subscriptionType;
+    }
+
+    /**
+     * @return ReferentTag[]
+     */
+    public function getReferentTags(): array
+    {
+        return $this->referentTags;
+    }
+
+    public function addReferentTag(ReferentTag $referentTag): void
+    {
+        $this->referentTags[] = $referentTag;
+    }
+
+    public function removeReferentTag(ReferentTag $referentTag): void
+    {
+        foreach ($this->referentTags as $key => $tag) {
```

Comment:
why do not use `$this->referentTags->removeElement` method?

Reply:
We don't need this method, I have been removed it.